core/matrix/dense.cpp

```diff
@@ -795,6 +798,81 @@ Dense<ValueType>::compute_absolute() const
 }
 
 
+template <typename ValueType>
+std::unique_ptr<typename Dense<ValueType>::complex_type>
+Dense<ValueType>::make_complex() const
+{
+    auto exec = this->get_executor();
+
+    auto complex_dense = complex_type::create(exec, this->get_size());
+
+    exec->run(dense::make_make_complex(this, complex_dense.get()));
+
+    return complex_dense;
+}
+
+
+template <typename ValueType>
+void Dense<ValueType>::make_complex(Dense<to_complex<ValueType>> *result) const
+{
+    auto exec = this->get_executor();
```

Comment:
`result` doesn't necessarily reside on the same executor as `this`, so in case the executors mismatch, you may need to do something like `result->copy_from(lend(this->get_real())`

Reply:
or a `make_temporary_clone`